### PR TITLE
Backport: Proxy X11 Forwarding Support

### PR DESCRIPTION
**Pull request by fspmarshall**
Backport of #3803

### CODE REVIEW

File: lib/srv/regular/sshserver_test.go [hunk 1]
```diff
@@ -1146,6 +1147,222 @@ func (s *SrvSuite) TestGlobalRequestRecordingProxy(c *C) {
...
+	addr := net.JoinHostPort(hostname, port)
+
+	return &rawNode{
+		listener: listener,
+		cfg:      cfg,
+		addr:     addr,
+	}
+}
+
+// startX11EchoServer starts a fake node which, for each incomging SSH connection, accepts an
```
Comment (webvictim) [inline lib/srv/regular/sshserver_test.go:1222]: incoming

File: lib/srv/forward/sshserver.go [hunk 2]
```diff
@@ -861,6 +863,186 @@ func (s *Server) handleAgentForward(ch ssh.Channel, req *ssh.Request, ctx *srv.S
...
+				}
+				continue
+			}
+		case <-ctx.Done():
+			return nil
+		}
+	}
+}
+
+// serveX11Channels registers and runs a hander for X11 channel requests
```
Comment (webvictim) [inline lib/srv/forward/sshserver.go:960]: handler

File: lib/srv/regular/sshserver_test.go [hunk 3]
```diff
@@ -1146,6 +1147,222 @@ func (s *SrvSuite) TestGlobalRequestRecordingProxy(c *C) {
...
+				sch, creqs, err := nch.Accept()
+				c.Assert(err, IsNil)
+				defer sch.Close()
+
+				// expect client to send an X11 forwarding request
+				var req *ssh.Request
+				select {
+				case req = <-creqs:
+				case <-time.After(time.Second * 3):
+					c.Fatalf("Timeout waiting for x11 forwarding request")
```
Comment (webvictim) [inline lib/srv/regular/sshserver_test.go:1259]: `X11`?

**Review by webvictim — APPROVED** [review]
Few nits I should have reviewed the first time round...

**Review by russjones — APPROVED** [review]

File: lib/srv/regular/sshserver_test.go (hunk 3 above)
Comment (fspmarshall) [reply lib/srv/regular/sshserver_test.go:1259]: This is deliberate.  I decided to follow the capitalization scheme that openssh uses for `X11`/`x11` (and what teleport uses for things like `HTTP`)... where it is lowercase in logs, errors, configs, etc.

**Pull request merged**